packages/javascript/bh-shared-ui/src/components/UpsertSSOProviders/UpsertOIDCProviderForm.tsx

```diff
@@ -18,25 +18,40 @@ import { Button } from '@bloodhoundenterprise/doodleui';
 import { Alert, DialogContent, DialogActions, Grid, TextField } from '@mui/material';
 import { FC } from 'react';
 import { Controller, useForm } from 'react-hook-form';
-import { OIDCProviderInfo, SSOProvider, UpsertOIDCProviderRequest } from 'js-client-library';
+import { OIDCProviderInfo, SSOProvider, UpsertOIDCProviderRequest, Role } from 'js-client-library';
+import SSOProviderConfigForm from './SSOProviderConfigForm';
 
 const UpsertOIDCProviderForm: FC<{
     error?: string;
     oldSSOProvider?: SSOProvider;
+    roles?: Role[];
     onClose: () => void;
     onSubmit: (data: UpsertOIDCProviderRequest) => void;
-}> = ({ error, oldSSOProvider, onClose, onSubmit }) => {
+}> = ({ error, oldSSOProvider, roles, onClose, onSubmit }) => {
+    const readOnlyRoleId = roles?.find((role) => role.name === 'Read-Only')?.id;
+
     const defaultValues = {
         name: oldSSOProvider?.name ?? '',
         client_id: (oldSSOProvider?.details as OIDCProviderInfo)?.client_id ?? '',
         issuer: (oldSSOProvider?.details as OIDCProviderInfo)?.issuer ?? '',
+        config: oldSSOProvider?.config
+            ? oldSSOProvider.config
+            : {
+                  auto_provision: {
```

Comment:
This appears to be a backfill for the config? Im thinking either a comment or named variable would help make that clearer